R/waapwlspetpeese.R

```diff
@@ -0,0 +1,610 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+### Common functions for WAAP-WLS and PET-PEESE
+.wwppDependencies <- c("measures", "muTransform", "inputES", "inputSE", "inputN")
+# check and load functions
+.wwppCheckReady              <- function(options) {
+  if (options[["measures"]] == "general") {
+    return(options[["inputES"]] != "" && options[["inputSE"]] != "")
+  } else if (options[["measures"]] == "correlation") {
+    return(options[["inputES"]] != "" && options[["inputN"]] != "")
+  }
+}
+.wwppGetData                 <- function(dataset, options) {
+  if (!is.null(dataset)) {
+    return(dataset)
+  } else {
+    return(.readDataSetToEnd(columns.as.numeric = c(
+      options[["inputES"]],
+      if (options[["inputSE"]] != "") options[["inputSE"]],
+      if (options[["inputN"]]  != "") options[["inputN"]]
+    )))
+  }
+}
+.wwppCheckData               <- function(jaspResults, dataset, options) {
+
+  dataset_old <- dataset
+  dataset     <- na.omit(dataset)
+
+  # store the number of missing values
+  if (nrow(dataset_old) > nrow(dataset)) {
+    if (!is.null(jaspResults[["nOmitted"]])) {
+      nOmitted <- jaspResults[["nOmitted"]]
+    } else {
+      nOmitted <- createJaspState()
+      nOmitted$dependOn(c(.wwppDependencies))
+      jaspResults[["nOmitted"]] <- nOmitted
+    }
+    nOmitted$object <- nrow(dataset_old) - nrow(dataset)
+  }
+
+  .hasErrors(dataset               = dataset,
+             type                  = c("infinity", "observations"),
+             observations.amount   = "< 2",
+             exitAnalysisIfErrors  = TRUE)
+
+  if (options[["inputSE"]] != "")
+    .hasErrors(dataset               = dataset,
+               type                  = c("negativeValues"),
+               negativeValues.target = options[["inputSE"]],
+               exitAnalysisIfErrors  = TRUE)
+
+  if (options[["inputN"]] != "")
+    .hasErrors(dataset               = dataset,
+               type                  = c("negativeValues"),
+               negativeValues.target = options[["inputN"]],
+               exitAnalysisIfErrors  = TRUE)
+
+
+  if (options[["inputES"]] != "" && options[["measures"]] == "correlation")
+    if (any(dataset[, options[["inputES"]]] <= -1) || any(dataset[, options[["inputES"]]] >= 1))
+      .quitAnalysis(gettextf("Error in %s: All correlation coefficients need to be between -1 and 1.", options[["inputES"]]))
```

Comment:
Maybe apply the same principle to this error message.

Reply:
done